patches/1.13.6.2/nginx-1.13.6_02-listen-transparent.patch

```diff
@@ -0,0 +1,213 @@
+diff -ru a/nginx-1.13.6/src/core/ngx_connection.c b/nginx-1.13.6/src/core/ngx_connection.c
+--- a/nginx-1.13.6/src/core/ngx_connection.c	2017-10-26 16:42:53.000000000 +1100
++++ b/nginx-1.13.6/src/core/ngx_connection.c	2018-08-02 15:22:53.652819986 +1000
+@@ -501,6 +501,49 @@
+             }
+ #endif
+
++#if (NGX_HAVE_TRANSPARENT_PROXY && __linux)
++            /* Don't need to set on FreeBSD or OSX */
++
++            if (ls[i].transparent && !ngx_test_config) {
++
++                int transparent = 1;
++
++                switch (ls[i].sockaddr->sa_family) {
++
++                case AF_INET:
++
++                    if (setsockopt(s, IPPROTO_IP, IP_TRANSPARENT,
++                                   (const void *) &transparent, sizeof(int)) == -1)
++                    {
++                        ngx_log_error(NGX_LOG_EMERG, log, ngx_socket_errno,
++                                      "setsockopt(IP_TRANSPARENT) %V failed",
++                                      &ls[i].addr_text);
++                        return NGX_ERROR;
++                    }
++
++                    break;
++
++#if (NGX_HAVE_INET6)
```

Comment:
May need to add `IPV6_TRANSPARENT` here. I recall running into a system without it when testing my patch further.